src/lsm/manifest_level.zig

```diff
@@ -54,6 +54,101 @@ pub fn ManifestLevelType(
             tables: std.BoundedArray(TableInfoReference, constants.lsm_growth_factor),
         };
 
+        const LevelKeyRange = struct {
+            key_min: ?Key = null,
+            key_max: ?Key = null,
```

Comment:
To understand, is there a specific reason that both of these are optional? i.e. Do we expect it to be possible that `key_min` can be null but not `key_max` or vice versa?

Why not make the struct itself optional? Then the owner of the memory can use null to check if the range is present or not, in a single branch instead of two.

Reply:
This could also use `kr: ?KeyRange` which keeps the owner from having to check/update/set on each update.

Reply:
We discussed this over call today as well! `key_min` & `key_max` would either both be `null` or `non-null`.

I shall make the `LevelKeyRange` itself optional!